main/src/library/Core/Functor.flix

```diff
@@ -0,0 +1,29 @@
+/*
+ *  Copyright 2020 Matthew Lutze
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+namespace Functor {
+    class Functor[m :# Type -> Type] {
```

Comment:
Scala:
```scala
trait Functor[F[_]] {
  def map[A, B](fa: F[A])(f: A => B): F[B]
}
```

Haskell:
```haskell
class Functor f where
    fmap :: (a -> b) -> f a -> f b
    (<$) :: a -> f b -> f a
```

Reply:
Is this just for comparison, or is there something you'd like to change?

Reply:
I just never noticed the difference before. In Haskell I assume the kind is inferred from the signatures (?) and what about Scala?

Reply:
I think Scala's kind is explicit here: `F[_]` is `Type -> Type`, `F[_, _]` would be `Type -> Type -> Type`

Reply:
OK; what is preferable? What is the most regular?

Reply:
In my opinion, the kind annotation works well because it parallels an arrow type annotation (e.g. `f: Int -> Int`).

Scala's version is nice and concise, but doesn't really fit with anything we've done so far.

We could do kind inference like in Haskell, but that could get messy fast.

Reply:
I have not yet formed an opinion. But if we go with kind annotations I think of two things:
- It would become a "Flix" thing-- something that visually sets the language apart. This could be both a strength or a weakness.
- This seems to reinforce that kinds should have a better syntax, either `:` or `<:` or `:<`, as we have talked about.

> In my opinion, the kind annotation works well because it parallels an arrow type annotation (e.g. f: Int -> Int).

I like this argument.

Anyway, this is not a discussion that has to be resolved now. Let us start with the kind annotations with the temporary syntax.